PathwayForte Resources/Results
==============================

Resources for the benchmarking of KEGG [1]_ [2]_ [3]_, Reactome [4]_ [5]_, and WikiPathways [6]_ [7]_ [8]_ on statistical enrichment and predictive Modeling using `PathwayForte <https://github.com/pathwayforte/pathway-forte>`_.

Structure
---------

- `Notebooks <https://github.com/pathwayforte/results/tree/master/notebooks/>`_. Jupyter notebooks showing the results of the analysis.
- `R <https://github.com/pathwayforte/results/tree/master/R>`_. Scripts to download, query and preprocessing TCGA datasets.
- `Input <https://github.com/pathwayforte/results/tree/master/input/>`_: Directories containing gmt files, TCGA gene expression data, enrichment scores and test data used in the paper
  (in case proprocessing has been already conducted)


References
----------
KEGG
~~~~

.. [1] Kanehisa, *et al.* (2017) KEGG: new perspectives on genomes, pathways, diseases and drugs. Nucleic Acids Res. 45,
       D353-D361.
.. [2] Kanehisa, M., *et al.* (2016). KEGG as a reference resource
       for gene and protein annotation. Nucleic Acids Res. 44, D457-D462.
.. [3] Kanehisa, M. and Goto, S. (2000). KEGG: Kyoto Encyclopedia of Genes and Genomes. Nucleic Acids Res. 28, 27-30.

Reactome
~~~~~~~~
.. [4] Fabregat, A *et al.* (2016). The Reactome Pathway Knowledgebase. Nucleic Acids Research 44. Database issue:
       D481–D487.
.. [5] Croft, D *et al.* (2014). The Reactome Pathway Knowledgebase. *Nucleic Acids Research* 42.Database issue:
       D472–D477.

WikiPathways
~~~~~~~~~~~~
.. [6] Slenter, D.N.,  *et al.* (2017). WikiPathways: a multifaceted pathway database bridging metabolomics to other omics
       research. *Nucleic Acids Research*, doi.org/10.1093/nar/gkx1064
.. [7] Kutmon, M., *et al.* (2016). WikiPathways: capturing the full diversity of pathway knowledge Nucl. Acids Res., 44,
       D488-D494.
.. [8] Kelder, T., *et al.* (201). WikiPathways: building research communities on biological pathways. Nucleic Acids Res.
       Jan;40(Database issue):D1301-7
